### PR TITLE
[WIP] Make COVID screener available in Spanish

### PR DESCRIPTION
**Pull request by batemapf**
## To do
- [x] Get updated translations for all content in /config/
- [x] Update/write tests
- [ ] User test with fluent / native Spanish speakers
- [x] Design review

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by rianfowler — COMMENTED** [review]
This might be a problem we have to confront on the platform at some point, so I'm curious if you considered or evaluated any third party packages or process for localization. We don't have to try to add that to this app but I think this is the first attempt to localize language in an app. 

cc @cvalarida 

- [react-localization](https://github.com/stefalda/react-localization#readme)
- [i18next](https://medium.com/@jishnu61/6-easy-steps-to-localize-your-react-application-internationalization-with-i18next-8de9cc3a66a1)

**Comment by batemapf** [comment]
@rianfowler @cvalarida This was definitely a quick and dirty effort to get basic Spanish-language functionality implemented. We talked quite a bit about approaches, though these discussions were mostly focused on these options:

- manual translation (as we did here);
- optimizing pages so they could be better translated by a client-side tool (e.g. the automatic translate feature in Chrome);
- programmatically translating content during build via an external service (e.g. the Google Translate API);
- some combination of the above.

The Public Websites team (cc @DanielleThierryUSDSVA @jenniferlee-dsva) have had a lot of conversations about which of those options (or others maybe) make sense for a larger effort. I've also suggested some investigation of platform-provided internationalization services to be added to the platform roadmap, but not sure where that ended up (cc @KevinMHoffmanUSDS @drorm @rroueche).

OK, to directly answer your question @rianfowler, no I did not investigate any packages that would help make this type manual implementation a bit easier. Hindsight being 20/20, that is an incredibly good idea. Just doing a quick [search](https://www.google.com/search?q=react+packages+for+internationalization) shows that there are plenty of things to consider. If the platform team does move down the internationalization road and you are looking for something to test out ideas on, happy to offer up this screener!

**Review by mchelen-gov — CHANGES_REQUESTED** [review]

File: src/applications/coronavirus-screener/components/FormQuestion.jsx [hunk 1]
```diff
@@ -41,14 +42,14 @@ export default function FormQuestion({
...
+      {option.optionText[selectedLanguage]}
     </button>
   ));
-
+  const questionText = question.text[selectedLanguage];
   return (
     <div className="feature" id={`question-${question.id}`}>
       <Element name={scrollElementName} />
-      <h2>{question.text}</h2>
+      <h2>{questionText}</h2>
```
[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/components/FormQuestion.jsx:52]: unnecessary temporary variable

File: src/applications/coronavirus-screener/config/text.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    en:
+      'Please answer the questions listed below. Share your results with the staff member at the facility entrance.',
+    es:
+      'Por favor, responda las preguntas que se enumeran a continuación. Comparta sus resultados con el miembro del personal que se encuentra en la entrada del establecimiento.',
+  },
+  privacytext: {
+    en: "We won't store or share your data.",
+    es: 'No guardaremos ni compartiremos sus datos.',
+  },
+  languageselecttext: {
```
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/config/text.jsx]: Updated this and other non-camels that were roaming around in the config files.

File: src/applications/coronavirus-screener/components/FormQuestion.jsx [hunk 3]
```diff
@@ -41,14 +42,14 @@ export default function FormQuestion({
       onClick={handleClick}
       value={option.optionValue}
     >
-      {option.optionText}
+      {option.optionText[selectedLanguage]}
     </button>
   ));
-
+  const questionText = question.text[selectedLanguage];
```
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/components/FormQuestion.jsx]: Cleaned up.

File: src/applications/coronavirus-screener/containers/App.jsx [hunk 4]
```diff
@@ -2,26 +2,53 @@ import React from 'react';
...
+import { introText } from '../config/text';
 
 export default function App({ params }) {
+  let selectedLanguage = 'en';
+  let alternateLangauge = 'es';
+  let alternateRefBase = '/covid19screen/';
+  let customId = params.id;
+
+  // If the first value is not a number, then there is no custom site id.
+  if (isNaN(params.id)) {
```
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/containers/App.jsx]: Add in a `.slice()` to just look at the the first three chars when determining if `NaN`. Admittedly, feels as brittle as the rest of this route-based implementation, but at least we have a variety of e2e scenarios that can validate expected behavior is happening.

File: src/applications/coronavirus-screener/components/FormResult.jsx [hunk 5]
```diff
@@ -3,44 +3,48 @@ import { Element } from 'react-scroll';
...
   );
 }
 
-const Pass = () => (
-  <Complete>
-    <i aria-hidden="true" role="presentation" className="fas fa-check" />
-    <h2 className="vads-u-font-size--2xl">OK to proceed</h2>
-  </Complete>
-);
+export default function FormResult({ formState, selectedLanguage }) {
```
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/components/FormResult.jsx:25]: Heh, can I plead ignorance? Still learning and would be happy to make this change in a refactor.

[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/components/FormResult.jsx:25]: yeah that's understandable, maybe add a `TODO` & create followup issue?

File: src/applications/coronavirus-screener/containers/App.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/containers/App.jsx]: yeah, dealing with strings this way can be a little unreliable. i think we know that all facility IDs are 3-digit numbers, right?

it seems like it should work properly, not sure if any E2E tests are confirming this currently

**Comment by batemapf** [comment]
Regarding the [disabled e2e tests](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/coronavirus-screener/tests/01.custom-questions.e2e.disabled.spec.js) for custom questions: I tried adding in a 2x longer pause to the scenario runner ([testQuestionScenario](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/coronavirus-screener/tests/question-scenario-helper.js#L6)) to see if that would do anything. No luck. Locally, it kept failing with: `Timed out while waiting for element <div[id=question-fever]> to be present for 5000 milliseconds`. Just to be sure it wasn't something weird going on with that div, I updated the scenario to remove the fever question. No luck there either, just failed on the next div.

So, TLDR, I'm at a loss for what is causing the test to hang at that particular part.

File: src/applications/coronavirus-screener/containers/App.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/containers/App.jsx]: E.g. https://github.com/department-of-veterans-affairs/vets-website/pull/14499/files#diff-a2056da60e50ae4479baf727309b3a23R118

[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/containers/App.jsx]: Yep all have a three digit base, but can have two letters following (like 459GH). In the real world, these signify community clinics that are associated with a medical center. So 459 is the Hawaii VAMC, and 459GH is the Hawaii VAMC clinic in the Northern Mariana Islands.

File: src/applications/coronavirus-screener/components/FormResult.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (batemapf) [reply src/applications/coronavirus-screener/components/FormResult.jsx:25]: done: https://github.com/department-of-veterans-affairs/vets-website/pull/14499/commits/b63f23b0f2ea0c422cfe9afe54ccb9cf2e5fdb94#diff-38951ffb31442c0171f7a5acc7eb6660R127

**Review by mchelen-gov — APPROVED** [review]
route test cases confirmed manually by @batemapf

File: src/applications/coronavirus-screener/components/FormResult.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/components/FormResult.jsx:25]: https://github.com/department-of-veterans-affairs/va.gov-team/issues/14691

File: src/applications/coronavirus-screener/containers/App.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/containers/App.jsx]: https://github.com/department-of-veterans-affairs/va.gov-team/issues/14692
